test/integration/common/test_console_output.py

```diff
@@ -0,0 +1,100 @@
+        happy_path_no_offset_1=(5, None, 5, 5, 10, "line_5\nline_6\nline_7\nline_8\nline_9\n"),
+        happy_path_no_offset_2=(1, None, 1, 9, 10, "line_9\n"),
+        out_of_lines_no_offset=(200, None, 10, 0, 10,
+                                "line_0\nline_1\nline_2\nline_3\nline_4\nline_5\nline_6\nline_7\nline_8\nline_9\n"),
+    )
+    def test_segment_for_completed_console_output(
+            self,
+            input_max_lines,
+            input_offset_line,
+            expected_num_lines,
+            expected_offset_line,
+            expected_total_num_lines,
+            expected_content
+    ):
+        """
+        :type input_max_lines: int
+        :type input_offset_line: int | None
+        :type expected_num_lines: int
+        :type expected_offset_line: int
+        :type expected_total_num_lines: int
+        :type expected_content: str
+        """
+        console_output = ConsoleOutput(self._completed_console_output_file_path)
+        segment = console_output.segment(max_lines=input_max_lines, offset_line=input_offset_line)
+
+        self.assertEquals(segment.num_lines, expected_num_lines)
+        self.assertEquals(segment.offset_line, expected_offset_line)
+        self.assertEquals(segment.total_num_lines, expected_total_num_lines)
+        self.assertEquals(segment.content, expected_content)
+
+    @genty_dataset(
+        mid_offset=(15, 5, 4, 5, 9, "line_5\nline_6\nline_7\nline_8\n"),
+        no_offset=(15, None, 9, 0, 9, "line_0\nline_1\nline_2\nline_3\nline_4\nline_5\nline_6\nline_7\nline_8\n"),
+        mid_lines_no_offset=(4, None, 4, 5, 9, "line_5\nline_6\nline_7\nline_8\n"),
+    )
+    def test_segment_for_incomplete_console_output(
+            self,
+            input_max_lines,
+            input_offset_line,
+            expected_num_lines,
+            expected_offset_line,
+            expected_total_num_lines,
+            expected_content
+    ):
+        """
+        :type input_max_lines: int
+        :type input_offset_line: int | None
+        :type expected_num_lines: int
+        :type expected_offset_line: int
+        :type expected_total_num_lines: int
+        :type expected_content: str
+        """
+        console_output = ConsoleOutput(self._incomplete_console_output_file_path)
+        segment = console_output.segment(max_lines=input_max_lines, offset_line=input_offset_line)
+
+        self.assertEquals(segment.num_lines, expected_num_lines)
+        self.assertEquals(segment.offset_line, expected_offset_line)
+        self.assertEquals(segment.total_num_lines, expected_total_num_lines)
+        self.assertEquals(segment.content, expected_content)
+
+    def test_segment_raises_value_error_if_offset_greater_than_total_length(self):
+        with self.assertRaises(ValueError):
+            console_output = ConsoleOutput(self._completed_console_output_file_path)
+            console_output.segment(max_lines=5, offset_line=155)
```

Comment:
I remember we ran into some weird issues with invalid unicode coming out of phpunit output that caused some exceptions. (We have this test around the issue: https://github.com/box/ClusterRunner/blob/master/test/unit/project_type/test_project_type.py#L78) I wonder if we should add some tests for this new functionality related to unicode console output and potentially malformed output.

I think we're a bit safer in this case because we're handling a client's API call when loading the file, so the worst that could happen is returning a 500 to the client.